src/moveit_visual_tools.cpp

```diff
@@ -1261,45 +1267,45 @@ bool MoveItVisualTools::publishTrajectoryPath(const moveit_msgs::RobotTrajectory
   // Wait the duration of the trajectory
   if (blocking)
   {
-    double duration = trajectory_msg.joint_trajectory.points.back().time_from_start.toSec();
+    double duration = trajectory_msg.joint_trajectory.points.back().time_from_start.sec;
 
     // If trajectory has not been parameterized, assume each waypoint takes 0.05 seconds (based on Rviz)
     if (duration < std::numeric_limits<double>::epsilon())
     {
       duration = 0.05 * trajectory_msg.joint_trajectory.points.size();
     }
-    ROS_DEBUG_STREAM_NAMED(LOGNAME, "Waiting for trajectory animation " << duration << " seconds");
+    RCLCPP_DEBUG_STREAM(LOGGER, "Waiting for trajectory animation " << duration << " seconds");
 
     // Check if ROS is ok in intervals
     double counter = 0;
     static const double CHECK_TIME_INTERVAL = 0.25;  // check every fourth second
-    while (ros::ok() && counter <= duration)
+    while (rclcpp::ok() && counter <= duration)
     {
       counter += CHECK_TIME_INTERVAL;
-      ros::Duration(CHECK_TIME_INTERVAL).sleep();
+      rclcpp::sleep_for(std::chrono::milliseconds(int(CHECK_TIME_INTERVAL * 1000)));
```

Comment:
```suggestion
      rclcpp::sleep_for(rclcpp::Duration::from_seconds(CHECK_TIME_INTERVAL));
```

Reply:
This does not compile, sleep_for cannot be initialized with `rclcpp::Duration::from_seconds`

Reply:
`std::chrono::seconds`?

Reply:
std::chrono::seconds takes integers only and I have 0.25 seconds to sleep.